File: zxingcamera/src/main/java/com/akingyin/zxingcamera/CameraSize.java
package com.akingyin.zxingcamera;

import android.graphics.Point;


/**
 * Name: CameraSize
 * Author: akingyin
 * Email:
 * Comment: //TODO
 * Date: 2019-04-10 11:35
 */
public class CameraSize   {



    private    int    width;

    private    int    hight;

    private    boolean   defalultSize;

    private    boolean   checked;


    public CameraSize(int width, int hight, boolean defalultSize, boolean checked) {
        this.width = width;
        this.hight = hight;
        this.defalultSize = defalultSize;
        this.checked = checked;
    }

    public CameraSize() {
    }

    public boolean isChecked() {
        return checked;
    }

    public void setChecked(boolean checked) {
        this.checked = checked;
    }

    public boolean isDefalultSize() {
        return defalultSize;
    }

    public void setDefalultSize(boolean defalultSize) {
        this.defalultSize = defalultSize;
    }

    public int getWidth() {
        return width;
    }

    public void setWidth(int width) {
        this.width = width;
    }

    public int getHight() {
        return hight;
    }

    public void setHight(int hight) {
        this.hight = hight;
    }


    @Override
    public String toString() {

        return CameraConfigurationUtils.getCameraResolutionScale(new Point(width,hight))+Math.max(width,hight)+"x"+Math.min(width,hight)+(defalultSize?"[默认]":"")+(checked?"[当前]":"");
    }
}

File: medialibs/src/main/java/com/akingyin/media/camera/CameraSize.kt
/*
 * Copyright (c) 2020. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
 * Morbi non lorem porttitor neque feugiat blandit. Ut vitae ipsum eget quam lacinia accumsan.
 * Etiam sed turpis ac ipsum condimentum fringilla. Maecenas magna.
 * Proin dapibus sapien vel ante. Aliquam erat volutpat. Pellentesque sagittis ligula eget metus.
 * Vestibulum commodo. Ut rhoncus gravida arcu.
 * akingyin@163.com
 */

package com.akingyin.media.camera

import android.graphics.Point
import kotlin.math.max
import kotlin.math.min


/**
 * @ Description:
 * @author king
 * @ Date 2020/7/29 16:16
 * @version V1.0
 */
data class CameraSize (var width:Int,var hight:Int,var defaultSize :Boolean= false,var checked:Boolean = false){
    override fun toString(): String {
        return CameraManager.getCameraResolutionScale(Point(width, hight)) + max(width, hight) + "x" + min(width, hight) + (if (defaultSize) "[默认]" else "") + if (checked) "[当前]" else ""
    }
}